js/client-battle-tooltips.js

```diff
@@ -37,11 +37,71 @@ var BattleTooltips = (function () {
 	};
 
 	// tooltips
-	BattleTooltips.prototype.tooltipAttrs = function (thing, type, ownHeight) {
-		var roomid = this.room.id;
-		return ' onmouseover="BattleTooltips.showTooltipFor(\'' + roomid + '\', \'' + Tools.escapeHTML('' + thing, true) + '\',\'' + type + '\', this, ' + (ownHeight ? 'true' : 'false') + ')" ontouchstart="BattleTooltips.showTooltipFor(\'' + roomid + '\', \'' + Tools.escapeHTML('' + thing, true) + '\',\'' + type + '\', this, ' + (ownHeight ? 'true' : 'false') + ')" onmouseout="BattleTooltips.hideTooltip()" onmouseup="BattleTooltips.hideTooltip()"';
-	};
+	if (navigator.userAgent.match(/(Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini)/)) {
+		// If user is using a mobile phone or tablet, use touch events
```

Comment:
Can't we use both, instead of branching here?

That also fixes the user agent matching, which, while sometimes necessary, isn't really necessary here.